packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -6,11 +6,54 @@ import 'dart:async';
 import 'dart:convert' show JSON;
 
 import 'package:crypto/crypto.dart' show md5;
+import 'package:meta/meta.dart';
 import 'package:quiver/core.dart' show hash3;
 
+import '../artifacts.dart';
 import '../build_info.dart';
+import '../globals.dart';
 import '../version.dart';
+import 'context.dart';
 import 'file_system.dart';
+import 'process.dart';
+
+GenSnapshot get genSnapshot => context.putIfAbsent(GenSnapshot, () => const GenSnapshot());
+
+/// A snapshot build configuration.
+class SnapshotType {
+  const SnapshotType(this.platform, this.mode);
+
+  final TargetPlatform platform;
+  final BuildMode mode;
+}
+
+/// Interface to the gen_snapshot command-line tool.
+class GenSnapshot {
+  const GenSnapshot();
+
+  Future<int> run({
+    @required SnapshotType snapshotType,
+    @required String packagesPath,
+    @required String depfilePath,
+    Iterable<String> additionalArgs: const <String>[],
+  }) {
+    final String vmSnapshotData = artifacts.getArtifactPath(Artifact.vmSnapshotData);
+    final String isolateSnapshotData = artifacts.getArtifactPath(Artifact.isolateSnapshotData);
+    final List<String> args = <String>[
+      '--assert_initializer',
+      '--await_is_keyword',
+      '--causal_async_stacks',
+      '--vm_snapshot_data=$vmSnapshotData',
+      '--isolate_snapshot_data=$isolateSnapshotData',
+      '--packages=$packagesPath',
+      '--dependencies=$depfilePath',
+      '--print_snapshot_sizes',
```

Comment:
can we remove this --print option? It adds spurious output when running `flutter run --release`.

Reply:
https://github.com/flutter/flutter/pull/18049